src/cli/cli.cpp

```diff
@@ -2348,40 +2350,41 @@ void Interpreter::ProcessRouter(int argc, char *argv[])
                 "+----+--------+----------+-----------+-------+--------+-----+------------------+\r\n");
         }
 
-        for (uint8_t i = 0;; i++)
+        maxRouterId = otThreadGetMaxRouterId(mInstance);
+
+        for (uint8_t i = 0; i <= maxRouterId; i++)
         {
-            if (otThreadGetRouterInfo(mInstance, i, &routerInfo) != OT_ERROR_NONE)
+            if ((otThreadGetRouterInfo(mInstance, i, &routerInfo) != OT_ERROR_NONE) || !routerInfo.mAllocated)
```

Comment:
Is `!routerInfo.mAllocated` necessary here? As in https://github.com/openthread/openthread/blob/master/src/core/thread/router_table.cpp#L458, `!routerInfo.mAllocated` would always be false if `OT_ERROR_NONE` is returned.